File: 2021/02/02a.swift
var instructions: [(String, Int)] = []
while let line = readLine() {
  let arr = line.split(separator: " ")
  instructions.append((String(arr[0]), Int(arr[1])!))
}

var horizontal = 0
var depth = 0
for (dir, dist) in instructions {
  if dir == "forward" {
    horizontal += dist
  } else if dir == "down" {
    depth += dist
  } else if dir == "up" {
    depth -= dist
  }
}

print(horizontal * depth)

File: 2021/02/02a.py
from sys import stdin

horizontal = 0
depth = 0
for dir, dist in ((dir, int(dist)) for dir, dist in map(lambda x: x.split(), stdin.readlines())):
    if dir == "forward":
        horizontal += dist
    elif dir == "down":
        depth += dist
    elif dir == "up":
        depth -= dist

print(horizontal * depth)
